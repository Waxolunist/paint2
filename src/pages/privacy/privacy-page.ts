import {
  css,
  CSSResult,
  customElement,
  html,
  LitElement,
  TemplateResult,
} from 'lit-element';
import {connect} from 'pwa-helpers/connect-mixin';
import store from '../../store';
import {arrowBack} from '../paint/icons';
import {navigate} from 'lit-redux-router';
import '../../components/icon-button/paint-icon-button';

@customElement('paint-privacy-page')
export class PrivacyPage extends connect(store)(LitElement) {
  render(): TemplateResult {
    return html`
      <section>
        <h2>Privacy policy</h2>
        <p>The app paint.v-collaborate.com is operated by Christian Sterzl.</p>
        <p>
          I collect information that your browser sends whenever you visit my
          Service ("Log Data"). This Log Data may include information such as
          your computer's Internet Protocol ("IP") address, browser type,
          browser version, the pages of our Service that you visit, the time and
          date of your visit, the time spent on those pages and other
          statistics.
          <br />
          The security of your Personal Information is important to me, but
          remember that no method of transmission over the Internet, or method
          of electronic storage is 100% secure. While I strive to use
          commercially acceptable means to protect your Personal Information, I
          cannot guarantee its absolute security.
          <br />
          The Service may contain links to other sites that are not operated by
          me. If you click on a third party link, you will be directed to that
          third party's site. I strongly advise you to review the Privacy Policy
          of every site you visit.
          <br />
          I have no control over, and assume no responsibility for the content,
          privacy policies or practices of any third party sites or services.
          <br />
          If you have any questions about this Privacy Policy, please contact me
          at christian.sterzl+paint[at]gmail.com.
        </p>
        <p class="back-button">
          <paint-icon-button
            class="back-button"
            @icon-clicked="${this.navigateBack}"
          >
            ${arrowBack}
          </paint-icon-button>
        </p>
      </section>
    `;
  }

  static get styles(): CSSResult {
    // language=CSS
    return css`
      :host {
        display: block;
        padding: 2em;
      }

      p.back-button {
        display: flex;
        justify-content: center;
      }

      paint-icon-button.back-button {
        --icon-size: 100px;
        width: 100px;
        height: 100px;
      }

      section {
        font-family: system-ui, sans-serif;
        font-size: large;
      }

      em {
        text-decoration: underline;
      }
    `;
  }

  private navigateBack(): void {
    store.dispatch(navigate('/'));
  }
}
