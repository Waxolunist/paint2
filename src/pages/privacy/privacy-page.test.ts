import {fixture} from '@open-wc/testing';
import {html} from 'lit-html';
import {PrivacyPage} from './privacy-page';

describe('privacy-page', () => {
  it('is defined', () => {
    const el = document.createElement('paint-privacy-page');
    expect(el).toBeInstanceOf(PrivacyPage);
  });

  it('renders correctly', async () => {
    const element = await fixture(
      html`<paint-privacy-page></paint-privacy-page>`
    );
    expect(element.shadowRoot!.innerHTML).toMatchSnapshot();
  });
});
