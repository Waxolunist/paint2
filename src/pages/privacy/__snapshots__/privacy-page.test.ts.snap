// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`privacy-page renders correctly 1`] = `
"<!---->
      <section>
        <h2>Privacy policy</h2>
        <p>The app paint.v-collaborate.com is operated by Christian Sterzl.</p>
        <p>
          I collect information that your browser sends whenever you visit my
          Service (\\"Log Data\\"). This Log Data may include information such as
          your computer's Internet Protocol (\\"IP\\") address, browser type,
          browser version, the pages of our Service that you visit, the time and
          date of your visit, the time spent on those pages and other
          statistics.
          <br>
          The security of your Personal Information is important to me, but
          remember that no method of transmission over the Internet, or method
          of electronic storage is 100% secure. While I strive to use
          commercially acceptable means to protect your Personal Information, I
          cannot guarantee its absolute security.
          <br>
          The Service may contain links to other sites that are not operated by
          me. If you click on a third party link, you will be directed to that
          third party's site. I strongly advise you to review the Privacy Policy
          of every site you visit.
          <br>
          I have no control over, and assume no responsibility for the content,
          privacy policies or practices of any third party sites or services.
          <br>
          If you have any questions about this Privacy Policy, please contact me
          at christian.sterzl+paint[at]gmail.com.
        </p>
        <p class=\\"back-button\\">
          <paint-icon-button class=\\"back-button\\">
            <svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 448 512\\">
  <path d=\\"M257.5 445.1l-22.2 22.2c-9.4 9.4-24.6 9.4-33.9 0L7 273c-9.4-9.4-9.4-24.6 0-33.9L201.4 44.7c9.4-9.4 24.6-9.4 33.9 0l22.2 22.2c9.5 9.5 9.3 25-.4 34.3L136.6 216H424c13.3 0 24 10.7 24 24v32c0 13.3-10.7 24-24 24H136.6l120.5 114.8c9.8 9.3 10 24.8.4 34.3z\\"></path>
</svg>
          </paint-icon-button>
        </p>
      </section>
    <!----><style>
      :host {
        display: block;
        padding: 2em;
      }

      p.back-button {
        display: flex;
        justify-content: center;
      }

      paint-icon-button.back-button {
        --icon-size: 100px;
        width: 100px;
        height: 100px;
      }

      section {
        font-family: system-ui, sans-serif;
        font-size: large;
      }

      em {
        text-decoration: underline;
      }
    </style>"
`;
